### ISSUE TITLE
VICE test drive/skew fails

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/virtualc64#384]
<img width="861" alt="diskskew" src="https://user-images.githubusercontent.com/12561945/43882015-7795558c-9bae-11e8-93b8-4774a5334199.png">

**Comment by dirkwhoffmann** [comment]
When running the test from the G64 image, it failed due to a bug in function Disk::encodeArchive(G64Archive *). When a track has size 0 in the image (which is usually the case for halftracks), the encoder assigned a default value. If the size is set equal to the previous (full) track, the test passes.

**Comment by dirkwhoffmann** [comment]
When running the test from the D64 image, the test passes if the (speed zone dependent) tail gaps 9, 12, 17, and 8 are used. I extracted these values from the Hoxs64 source code.

Formerly, I used 9,9,9,9 as tail gaps. I extracted these values from a G64 image that I created with 64COPY. 

As usual, I have no clue what VICE is doing. The compatibility of VICE is awesome (as well as the test suite the VICE team produced), but their source code is 😲.

I'll use the Hoxs64 values from now on. Interestingly, Hoxs64 fails on this test although it uses the correct "magic" tail gap values.
